datafusion/src/physical_plan/expressions/in_list.rs

```diff
@@ -234,16 +363,40 @@ impl PhysicalExpr for InListExpr {
 
         match value_data_type {
             DataType::Float32 => {
-                make_contains!(array, list_values, self.negated, Float32, Float32Array)
+                make_contains_primitive!(
+                    array,
+                    list_values,
+                    self.negated,
+                    Float32,
+                    Float32Array
+                )
             }
             DataType::Float64 => {
-                make_contains!(array, list_values, self.negated, Float64, Float64Array)
+                make_contains_primitive!(
+                    array,
+                    list_values,
+                    self.negated,
+                    Float64,
+                    Float64Array
+                )
             }
             DataType::Int16 => {
-                make_contains!(array, list_values, self.negated, Int16, Int16Array)
+                make_contains_primitive!(
+                    array,
+                    list_values,
+                    self.negated,
+                    Int16,
+                    Int16Array
+                )
             }
             DataType::Int32 => {
-                make_contains!(array, list_values, self.negated, Int32, Int32Array)
+                make_contains_primitive!(
+                    array,
+                    list_values,
+                    self.negated,
+                    Int32,
+                    Int32Array
+                )
             }
             DataType::Int64 => {
                 make_contains!(array, list_values, self.negated, Int64, Int64Array)
```

Comment:
Is there a reason not to apply the same transformation to `Int64` and `Int8`?

Reply:
Oops, great catch.